compat/block-editor/widget-block.php

```diff
@@ -8,40 +8,70 @@ class SiteOrigin_Widgets_Bundle_Widget_Block {
 	 */
 	public static function single() {
 		static $single;
-		
+
 		return empty( $single ) ? $single = new self() : $single;
 	}
-	
+
 	public function __construct() {
 		add_action( 'init', array( $this, 'register_widget_block' ) );
 		add_action( 'enqueue_block_editor_assets', array( $this, 'enqueue_widget_block_editor_assets' ) );
 	}
-	
+
 	public function register_widget_block() {
 		register_block_type( 'sowb/widget-block', array(
 			'render_callback' => array( $this, 'render_widget_block' ),
 		) );
 	}
-	
+
 	public function enqueue_widget_block_editor_assets() {
 		wp_enqueue_script(
 			'sowb-widget-block',
 			plugins_url( 'widget-block' . SOW_BUNDLE_JS_SUFFIX . '.js', __FILE__ ),
 			array( 'wp-editor', 'wp-blocks', 'wp-i18n', 'wp-element', 'wp-components', 'wp-compose' ),
 			SOW_BUNDLE_VERSION
 		);
-		
+
+		$widgets_metadata_list = SiteOrigin_Widgets_Bundle::single()->get_widgets_list();
+		$widgets_manager = SiteOrigin_Widgets_Widget_Manager::single();
+
 		global $wp_widget_factory;
 		$so_widgets = array();
+		$third_party_widgets = array();
 		foreach ( $wp_widget_factory->widgets as $class => $widget_obj ) {
 			if ( ! empty( $widget_obj ) && is_object( $widget_obj ) && is_subclass_of( $widget_obj, 'SiteOrigin_Widget' ) ) {
-				$so_widgets[] = array(
-					'name' => preg_replace( '/^SiteOrigin /', '', $widget_obj->name ),
-					'class' => $class,
-				);
+				/** @var SiteOrigin_Widget $widget_obj */
+				$author = '';
+				// Try to find a widget's author from it's file metadata, by matching the filename to the ID (which is derived from the filename).
+				foreach ( $widgets_metadata_list as $widget_metadata ) {
+					$filename = $widgets_manager->get_widget_filename( $widget_obj->id_base );
+					if ( $widget_metadata['ID'] == $filename ) {
+						$author = $widget_metadata['Author'];
+						break;
+					}
+				}
+				// For SiteOrigin widgets, just display the widget's name. For third party widgets, display the Author
+				// to try avoid confusion when the widgets have the same name.
+				if ( preg_match( '/^SiteOrigin /', $widget_obj->name ) == 1 && $author == 'SiteOrigin' ) {
+					$name = preg_replace( '/^SiteOrigin /', '', $widget_obj->name );
+
+					$so_widgets[] = array(
+						'name' => $name,
+						'class' => $class,
+					);
+				} else {
+					$name = sprintf( __( '%s by %s', 'so-widgets-bundle' ), $widget_obj->name, $author );
+					$third_party_widgets[] = array(
+						'name' => $name,
+						'class' => $class,
+					);
+				}
 			}
 		}
-		
+		// Sort the list of widgets so SiteOrigin widgets are at the top and then third party widgets.
+		sort($so_widgets);
```

Comment:
Minor formatting change here if possible. `sort($so_widgets);` to `sort( $so_widgets );`. Thanks.